### ISSUE TITLE
DataTestMethod skips DataRows that provide different data arrays, only keeping last

### ISSUE THREAD

**Issue by sandersaares** [microsoft/testfx#1016]
## Steps to reproduce
```csharp
[DataTestMethod]
[DataRow(0, new int[] { })]
[DataRow(0, new int[] { 0 })]
[DataRow(0, new int[] { 0, 0, 0 })] // Is executed
[DataRow(1, new int[] { 1 })]
[DataRow(1, new int[] { 0, 1 })] // Is executed
public void TestMethod1(int expectedSum, int[] buckets)
{
    Assert.AreEqual(expectedSum, buckets.Sum());
}
```

## Expected behavior
Test is executed with 5 sets of arguments.

## Actual behavior
Test is executed with only 2 sets of arguments (see comments in snippet).

![image](https://user-images.githubusercontent.com/9914262/144189671-7f4d40bc-fb37-4062-adb1-0f968da99149.png)

Test discovery DOES see all 5 test cases:

```
========== Starting test discovery ==========
========== Test discovery finished: 5 Tests found in 450,8 ms ==========
========== Starting test run ==========
========== Test run finished: 2 Tests (2 Passed, 0 Failed, 0 Skipped) run in 69 ms ==========
```

## Environment
Windows 11, Visual Studio 2022 v17.0.1, .NET SDK 6.0.100.

```xml
  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.11.0" />
    <PackageReference Include="MSTest.TestAdapter" Version="2.2.7" />
    <PackageReference Include="MSTest.TestFramework" Version="2.2.7" />
    <PackageReference Include="coverlet.collector" Version="3.1.0" />
  </ItemGroup>
```

**Comment by notcake** [comment]
I suspect that all `DataRow` tests with the same display name end up with the same GUID, and so only one survives:

https://github.com/microsoft/testfx/blob/57d2d00637e2bb3b5b9c4f9334a10ef2f6463146/src/Adapter/MSTest.CoreAdapter/ObjectModel/UnitTestElement.cs#L232-L252

Note line 247, where `testCase.DisplayName` is fed to `TestIdProvider`, which generates the GUID.

Untested, but I think this should be:
```diff
              if (this.TestMethod.DataType != DynamicDataType.None)
              {
                  idProvider.AppendString(testCase.DisplayName);
+                 foreach (string serializedData in this.TestMethod.SerializedData)
+                 {
+                     idProvider.AppendString(serializedData);
+                 }
              }
```

**Comment by Haplois** [comment]
That's possible solution @notcake, but it breaks compatibility. 

@sandersaares, could you try our latest version, this issue should be fixed in it.

**Comment by Evangelink** [comment]
We are working on the fix as I write these lines. I just wanted to mention/keep track of the fact that doing a test with `Microsoft.NET.Test.Sdk` version `17.3.2` and MSTest version `2.2.10` I can correctly reproduce the issue in VS (only 2 tests appear) but result is correct in command line (5 tests appear).

**Comment by Evangelink** [comment]
Fixed through #1306
